app/code/Magento/Ui/view/base/web/js/grid/masonry.js

```diff
@@ -0,0 +1,280 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+define([
+    'Magento_Ui/js/grid/listing',
+    'jquery',
+    'ko'
+], function (Listing, $, ko) {
+    'use strict';
+
+    return Listing.extend({
+        defaults: {
+            template: 'Magento_Ui/grid/masonry',
+            imports: {
+                rows: '${ $.provider }:data.items',
+                errorMessage: '${ $.provider }:data.errorMessage'
+            },
+            listens: {
+                'rows': 'initComponent'
+            },
+
+            /**
+             * Images container id
+             * @param string
+             */
+            containerId: null,
+
+            /**
+             * Minimum aspect ratio for each image
+             * @param int
+             */
+            minRatio: null,
+
+            /**
+             * Container width
+             * @param int
+             */
+            containerWidth: window.innerWidth,
+
+            /**
+             * Margin between images
+             * @param int
+             */
+            imageMargin: 20,
+
+            /**
+             * Maximum image height value
+             * @param int
+             */
+            maxImageHeight: 240,
+
+            /**
+             * The value is minimum image width to height ratio when container width is less than the key
+             * @param {Object}
+             */
+            containerWidthToMinRatio: {
+                640: 3,
+                1280: 5,
+                1920: 8
+            },
+
+            /**
+             * Default minimal image width to height ratio.
+             * Applied when container width is greater than max width in the containerWidthToMinRatio matrix.
+             * @param int
+             */
+            defaultMinRatio: 10
+        },
+
+        /**
+         * Init observable variables
+         * @return {Object}
+         */
+        initObservable: function () {
+            this._super()
+                .observe([
+                    'rows',
+                    'errorMessage'
+                ]);
+
+            return this;
+        },
+
+        /**
+         * Init component handler
+         * @param {Object} rows
+         * @return {Object}
+         */
+        initComponent: function (rows) {
```

Comment:
This method's docblock says it'll be receiving an object, but the first line of code does a safety check for if `rows` is truthy. We should only do a length check here, and callers should not be invoking `initComponent` with undefined or null

Reply:
Done